### ISSUE TITLE
Flask app reload raises exception on repeat calls to enable_attach

### ISSUE THREAD

**Issue by bitstrider** [Microsoft/ptvsd#1152]
I'm trying to integrate **ptvsd** with a **flask** app that has reloading feature enabled (as in it restarts when files are changed). This causes the ptvsd boilerplate code to be re-ran each time a reload happens, which results in a raised exception on repeat calls to `ptvsd.enable_attach`:

## Environment data

- PTVSD version: **4.2.3**
- OS and version: **Windows 10 Pro 1703**
- Python version (& distribution if applicable, e.g. Anaconda): **3.6.8**
- Using VS Code or Visual Studio: **VS Code**

## Actual behavior

raises exception on repeat calls to `ptvsd.enable_attach`:

```
OSError(10048, 'Only one usage of each socket address (protocol/network address/port) is normally permitted', None, 10048, None)
```

afterwards attaching to debugger works but does not work properly (breakpoints stop getting caught)

## Expected  behavior

handle reloading gracefully.

## Steps to reproduce:
check out this repo which demos this issue

1. `git clone https://github.com/bitstrider/ptvsd-flask-reload.git`
2. `pip install -r requirements.txt`
3. open VS code and add project to workspace
4. `./run.sh`
5. wait for error to print to terminal

**Comment by karthiknadig** [comment]
@bitstrider You will need #1153 for this to work correctly. But, there are some changes to how you launch flask using ptvsd.
1. you don't need to call `ptvsd.enable_attach` if you are already using `ptvsd` to launch `flask` module in `run.sh`.
**app.py**:
```py
from flask import Flask

app = Flask(__name__)

@app.route('/')
def hello_world():
    return 'Hello, VSCode!'
```

2. Use `--multiprocess` flag and `--wait` flags. `--multiprocess` flag enables subprocess tracking in `ptvsd`. You will need wait because, `flask` may start the child process before you can connect to the main process. VSC will not be able to connect to already started sub-process. Running the debugger with wait will block flask from running until you connect to the debugger.
`-m ptvsd --host localhost --port 5678 --multiprocess --wait -m flask run --host=localhost --no-debugger`


3. Use this as attach configuration in VSC:
```js
        {
            "name": "Flask (remote)",
            "type": "python",
            "request": "attach",
            "port": 5678,
            "host": "localhost",
            "jinja": true, // for template debugging
            "subProcess": true, // for sub-process debugging
            "pathMappings": [
                {
                    "localRoot": "${workspaceFolder}",
                    "remoteRoot": "."
                },
            ]
        }
```

That will setup VSC to automatically attach to child process. 
![image](https://user-images.githubusercontent.com/3840081/52775768-499c8c00-2ff5-11e9-82c0-e13a577f1ed4.png)


This is with the fix in #1153.

**Comment by bitstrider** [comment]
hey @karthiknadig thanks for all that, looking forward to the release 💯
